Towny/src/main/java/com/palmergames/bukkit/towny/command/TownyAdminCommand.java

```diff
@@ -1840,35 +1852,24 @@ private void adminSet(CommandSender sender, String[] split) throws TownyExceptio
 			return;
 		}
 
-		switch (split[0].toLowerCase(Locale.ROOT)) {
-		case "capital":
-			adminSetCapital(sender, split);
-			break;
-		case "founder":
-			adminSetFounder(sender, split);
-			break;
-		case "mayor":
-			adminSetMayor(sender, split);
-			break;
-		case "nationzoneoverride":
-			adminSetNationZoneOverride(sender, split);
-			break;
-		case "plot":
-			adminSetPlot(sender, split);
-			break;
-		case "surname":
-			adminSetSurname(sender, split);
-			break;
-		case "title":
-			adminSetTitle(sender, split);
-			break;
-		default:
-			if (TownyCommandAddonAPI.hasCommand(CommandType.TOWNYADMIN_SET, split[0])) {
-				TownyCommandAddonAPI.getAddonCommand(CommandType.TOWNYADMIN_SET, split[0]).execute(sender, "townyadmin", split);
-			} else {
-				TownyMessaging.sendErrorMsg(sender, Translatable.of("msg_err_invalid_property", "administrative"));
-			}
-		}
+        switch (split[0].toLowerCase(Locale.ROOT)) {
+            case "capital" -> adminSetCapital(sender, split);
+            case "founder" -> adminSetFounder(sender, split);
+            case "mayor" -> adminSetMayor(sender, split);
+            case "nationzoneoverride" -> adminSetNationZoneOverride(sender, split);
+            case "plot" -> adminSetPlot(sender, split);
+            case "surname" -> adminSetSurname(sender, split);
+            case "title" -> adminSetTitle(sender, split);
+            case "townfoundingdate" -> adminSetTownFoundingDate(sender, split);
+			case "nationfoundingdate" -> adminSetNationFoundingDate(sender, split);

```

Comment:
I think it would be best to have these as part of the ta town townname set/ta nation nationname set commands instead of in the root set command

Reply:
At the moment ```/ta town set``` is used to set the properties a mayor can set using ```/town set```, the tab complete is even hotwired to go there. We don't want the logic for a townyadmin command going into the TownCommand.

Reply:
done